compiler/passes/scopeResolve.cpp

```diff
@@ -467,6 +483,15 @@ static void scopeResolve(const AList& alist, ResolveScope* scope) {
         }
       }
 
+      // Look for IfExprs
+      if (def->init != NULL) {
```

Comment:
Hmm, the compiler doesn't emit an error for that pattern. I'll investigate.

Reply:
Perhaps unsurprisingly, your example doesn't work correctly because normalize is making decisions based on x's type (known to be a primitive, int). If we leave off the type expression then we correctly catch the use-before-def.